MekHQ/src/mekhq/campaign/io/CampaignXmlParser.java

```diff
@@ -413,7 +413,9 @@ else if (xn.equalsIgnoreCase("info")) {
                         continue;
                     }
                     // Remove existing duplicate parts.
-                    if (u.getPartForEquipmentNum(((EquipmentPart) prt).getEquipmentNum(), ((EquipmentPart) prt).getLocation()) != null) {
+                    Part duplicatePart = u.getPartForEquipmentNum(((EquipmentPart) prt).getEquipmentNum(), ((EquipmentPart) prt).getLocation());
+                    if (duplicatePart instanceof EquipmentPart
```

Comment:
I assume that duplicatePart instanceof EquipmentPart will return false rather than throwing an NPE when duplicatePart is null?

Reply:
Null check would be redundant per the spec: "At run time, the result of the instanceof operator is true if the value of the RelationalExpression is not null and the reference could be cast to the ReferenceType without raising a ClassCastException. Otherwise the result is false."